Automattic/jetpack
# Protect: Add Scan History UI

## Description
Adds Scan History and surrounding data handling and state management to the Protect Scan UI.

## Proposed changes:

* Adds a `viewingScanHistory` property to the initial state
* Adds a `useScanHistory` hook for handling of scan archives
* Uses the `useScanHistory` hook in the `useProtectData` hook to conditionally supply archives to the `ScanPage` components when `viewScanHistory` is toggled
* Refactor and optimizes the `ScanPage` and its various states

### Other information:

## Jetpack product discussion


- https://github.com/Automattic/jetpack-scan-team/issues/1225

## Does this pull request change what data or activity we track or use?


- No

## Testing instructions:






* Checkout this branch
* Fire up Jurassic Tube, enable Protect
* Add various threats for each type and ensure the site your using has archives
   * Verify using the wpcom Dev API console making a `GET` request to `/sites/<site_id>/scan/history`
* Ensure the Protect report displays as per usual, and you can toggle through list without issue
* Ensure nothing related to the history is displayed or found within the initial state
* Upgrade and run a scan to detect the current threats
* Ensure the Scan status report displays as per usual, and you can toggle through list without issue
* Verify that in each state of the `ScanPage` (error/unavailable, scanning, idle) the `History`/`Current` button is displayed
* Toggle to and from the History and verify that the list reflects the same details Calypso displays
* Fix and/or ignore a threat and ensure the History updates with the status
* Ensure no regressions in functionality are introduced

## Screenshots
Scan - "error/unavailable"
![Screen Shot on 2024-06-25 at 15-13-49](https://github.com/Automattic/jetpack/assets/43220201/938e2482-6524-433c-b2bb-9ec1780ed42e)

Scan - "scanning"
![Screen Shot on 2024-06-25 at 15-12-32](https://github.com/Automattic/jetpack/assets/43220201/638bdced-2c95-4ee8-901e-91e2c5d05fb5)

Scan - "idle"
![Screen Shot on 2024-06-25 at 15-11-53](https://github.com/Automattic/jetpack/assets/43220201/73562047-085f-48e3-b788-20fdf7384639)

History - "error"
![Screen Shot on 2024-06-25 at 15-14-11](https://github.com/Automattic/jetpack/assets/43220201/28177c43-397b-4296-875a-22be2de3c553)

History - "idle"
![Screen Shot on 2024-06-25 at 15-14-55](https://github.com/Automattic/jetpack/assets/43220201/0f8ae410-23da-4d37-b993-f0328180d1c1)
